lib/zlib.js

```diff
@@ -361,11 +361,10 @@ ZlibBase.prototype._final = function(callback) {
 // Z_NO_FLUSH (< Z_TREES) < Z_BLOCK < Z_PARTIAL_FLUSH <
 //     Z_SYNC_FLUSH < Z_FULL_FLUSH < Z_FINISH
 const flushiness = [];
-let i = 0;
 const kFlushFlagList = [Z_NO_FLUSH, Z_BLOCK, Z_PARTIAL_FLUSH,
                         Z_SYNC_FLUSH, Z_FULL_FLUSH, Z_FINISH];
-for (const flushFlag of kFlushFlagList) {
-  flushiness[flushFlag] = i++;
+for (let i = 0; i < kFlushFlagList.length; i++) {
+  flushiness[kFlushFlagList[i]] = i;
 }
```

Comment:
Can we use [for...in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) loop?
```suggestion
for (const i in kFlushFlagList.length) {
  flushiness[kFlushFlagList[i]] = i;
}
```

Reply:
Just read the primordial docs.

The for..in will be as bad as for..of loop as explained in the first point
https://github.com/nodejs/node/blob/main/doc/contributing/primordials.md?rgh-link-date=2024-08-07T23%3A34%3A39Z#unsafe-array-iteration

Reply:
@trivikr 
Even so, thank you for the advice.